src/commands/account/create_account/fund_myself_create_account/mod.rs

```diff
@@ -0,0 +1,246 @@
+                        MIN_ALLOWED_TOP_LEVEL_ACCOUNT_LENGTH,
+                    );
+                    if !is_input_new_name()? {
+                        break new_account_id;
+                    };
+                } else {
+                    let parent_account_id = new_account_id
+                        .clone()
+                        .get_owner_account_id_from_sub_account();
+                    if !near_primitives::types::AccountId::from(parent_account_id.clone())
+                        .is_top_level()
+                    {
+                        if optional_new_account_view(context, parent_account_id.clone().into())?
+                            .is_none()
+                        {
+                            println!("\nThe parent account <{}> does not yet exist. Therefore, you cannot create an account <{}>.",
+                            &parent_account_id, &new_account_id);
+                            if !is_input_new_name()? {
+                                break new_account_id;
+                            };
+                        } else {
+                            break new_account_id;
+                        }
+                    } else {
+                        break new_account_id;
+                    }
+                };
+                new_account_id = Input::new()
+                    .with_prompt("What is the new account ID?")
+                    .interact_text()?;
+            }
+        } else {
+            new_account_id
+        };
+        Ok(account_id)
+    }
+
+    fn input_initial_balance(
+        _context: &crate::GlobalContext,
+    ) -> color_eyre::eyre::Result<crate::common::NearBalance> {
+        println!();
+        let initial_balance: crate::common::NearBalance = Input::new()
+            .with_prompt(
+                "Enter deposit for a function call (example: 10NEAR or 0.5near or 10000yoctonear).",
+            )
+            .with_initial_text("0.1 NEAR")
+            .interact_text()?;
+        Ok(initial_balance)
+    }
+
+    pub async fn process(&self, config: crate::config::Config) -> crate::CliResult {
+        let account_properties = AccountProperties {
+            new_account_id: self.new_account_id.clone().into(),
+            initial_balance: self.initial_balance.clone(),
+            public_key: near_crypto::PublicKey::empty(near_crypto::KeyType::ED25519),
+        };
+        self.access_key_mode
+            .process(config, account_properties)
+            .await
+    }
+}
+
+#[derive(Debug, Clone)]
+struct NewAccountView {
+    _optional_account_view: Option<near_primitives::views::AccountView>,
+    network_config: crate::config::NetworkConfig,
+}
+
+fn optional_new_account_view(
+    context: &crate::GlobalContext,
+    new_account_id: near_primitives::types::AccountId,
+) -> color_eyre::eyre::Result<Option<NewAccountView>> {
+    for network in context.0.networks.iter() {
+        let _is_new_account_id = loop {
```

Comment:
Given that this code was extracted into a separate function, we don't need the guard that I explained before.

```suggestion
        loop {
```

Also, replace `break false;` with just `break;`

Reply:
done